Update org.reflections:reflections dependency to new version 0.10.X

from https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=5737&view=logs&j=1a6076f9-5826-5ba0-19ba-deaff2e35eea&t=5f46bc98-6f46-512e-6656-5fbfb1d03e4d&l=595

Update reflections dependency to new version.
Remove workaround that we did at https://github.com/checkstyle/checkstyle/pull/10902

UT failure:
```
~/java/github/romani/checkstyle [master|✚ 1] 
$ git diff
diff --git a/pom.xml b/pom.xml
index 43c2c50..7b36d2d 100644
--- a/pom.xml
+++ b/pom.xml
@@ -261,7 +261,7 @@
     <dependency>
       <groupId>org.reflections</groupId>
       <artifactId>reflections</artifactId>
-      <version>0.9.12</version>
+      <version>0.10.1</version>
     </dependency>
 
     <!-- test scope stuff -->

$ mvn test
....
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   XmlMetaReaderTest.test:49 » NullPointer
[ERROR]   XmlMetaReaderTest.testBadPackage:61 » NullPointer
[ERROR]   XmlMetaReaderTest.testDuplicatePackage:55 » NullPointer
[INFO] 
[ERROR] Tests run: 3646, Failures: 0, Errors: 3, Skipped: 0
```

--- a/pom.xml
+++ b/pom.xml
@@ -261,7 +261,7 @@
     <dependency>
       <groupId>org.reflections</groupId>
       <artifactId>reflections</artifactId>
-      <version>0.9.12</version>
+      <version>0.10.1</version>
     </dependency>
 
     <!-- test scope stuff -->

--- a/src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java
@@ -33,8 +33,7 @@ import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.parsers.ParserConfigurationException;
 
 import org.reflections.Reflections;
-import org.reflections.ReflectionsException;
-import org.reflections.scanners.ResourcesScanner;
+import org.reflections.scanners.Scanners;
 import org.w3c.dom.Document;
 import org.w3c.dom.Element;
 import org.w3c.dom.NodeList;
@@ -73,22 +72,15 @@ public final class XmlMetaReader {
      */
     public static List<ModuleDetails> readAllModulesIncludingThirdPartyIfAny(
             String... thirdPartyPackages) {
-        final Set<String> standardModuleFileNames =
-                new Reflections("com.puppycrawl.tools.checkstyle.meta",
-                        new ResourcesScanner()).getResources(Pattern.compile(".*\\.xml"));
+        final Set<String> standardModuleFileNames = new Reflections(
+                "com.puppycrawl.tools.checkstyle.meta", Scanners.Resources)
+                .getResources(Pattern.compile(".*\\.xml"));
         final Set<String> allMetadataSources = new HashSet<>(standardModuleFileNames);
         for (String packageName : thirdPartyPackages) {
-            try {
-                final Set<String> thirdPartyModuleFileNames =
-                        new Reflections(packageName, new ResourcesScanner())
-                                .getResources(Pattern.compile(".*checkstylemeta-.*\\.xml"));
-                allMetadataSources.addAll(thirdPartyModuleFileNames);
-            }
-            catch (ReflectionsException ex) {
-                if (propagateResourceScannerException(ex)) {
-                    throw ex;
-                }
-            }
+            final Set<String> thirdPartyModuleFileNames =
+                    new Reflections(packageName, Scanners.Resources)
+                            .getResources(Pattern.compile(".*checkstylemeta-.*\\.xml"));
+            allMetadataSources.addAll(thirdPartyModuleFileNames);
         }
 
         final List<ModuleDetails> result = new ArrayList<>();
@@ -116,20 +108,6 @@ public final class XmlMetaReader {
         }
 
         return result;
-    }
-
-    /**
-     * Checks if exception should be suppressed that matches the following
-     * because the reflections library throws this exception wrongly on valid
-     * packages with no resources to be found. Packages may be only checks with
-     * no resources or checks with resources. This issue is until
-     * https://github.com/ronmamo/reflections/issues/273.
-     *
-     * @param exception The exception to examine.
-     * @return {@code true} if the exception should be propagated.
-     */
-    private static boolean propagateResourceScannerException(ReflectionsException exception) {
-        return !"Scanner ResourcesScanner was not configured".equals(exception.getMessage());
     }
 
     /**

